src/operator/tensor/broadcast_reduce-inl.cuh

```diff
@@ -60,16 +60,18 @@ __global__ void reduce_kernel(const int N, const int M, const bool addto,
           for (int u=0;u < unroll;u++) {
             idx_big[u] = idx_big0 + mxnet_op::unravel_dot(k + u*by, big_shape, big_stride);
           }
-          DType tmp[unroll];
+          AType tmp[unroll];
```

Comment:
Why change DType --> AType?

Reply:
I though this was a little cleaner. This way we can get rid of the explicit conversion on line 74

Reply:
I think this is for the purpose of caching these data to the local register. Basically, we load `unroll` elements to the local register and then accumulate them to the accumulation type.

Reply:
Another reason is that the return type of OP::Map()  is `struct IndexedNum`. This would be the least changes to make the kernel compatible with that. I am trying to think of another solution

Reply:
It's probably not a good idea to add too much branching in the gpu kernel... If there is no alternative solution I can write a custom GPU kernel. This way we can also get rid of the residual variable which we is not using for arg min max

Reply:
I just compared my pr build and master build. Same config: Cuda on, large tensor on.

I checked np.sum and used mxnet profiler and this change does not seem to affect caching